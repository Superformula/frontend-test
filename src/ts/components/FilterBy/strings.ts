export default {
    clearAll: "Clear All"
};
