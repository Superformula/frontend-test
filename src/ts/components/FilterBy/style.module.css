.filterBy {
    display: flex;
    align-items: center;
    user-select: none;
}

.filterByText {
    margin-right: 24px;
}

.filterByOption {
    margin-right: 33px;
}

.clearAllButton {
    display: flex;
    justify-content: flex-end;
    flex-grow: 1;
}
