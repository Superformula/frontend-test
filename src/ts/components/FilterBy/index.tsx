// Imports
// -------

// Libraries
import React from 'react';
// Components
import Button from '../Button';
// Strings
import strings from "./strings";
// Style
import style from './style.module.css';

// Internal
// --------

export interface FilterByProps {
    children: React.ReactNode,
    clearAll?(): any,
    disableClearAll?: boolean
}

const FilterBy: React.FC<FilterByProps> = ({children, clearAll = () => {}, disableClearAll = false}) => {
    //
    // Render

    return (
        <div className={ style.filterBy }>
            <div className={ style.filterByText }>Filter By:</div>
            {
                React.Children.map(children, (child, idx) => (
                    <div key={idx} className={ style.filterByOption }>
                        { child }
                    </div>
                ))
            }
            <div className={ style.clearAllButton }>
                <Button text={ strings.clearAll } tier="secondary" onClick={ clearAll } disabled={ disableClearAll }/>
            </div>
        </div>
    );
}

// Exports
// -------

export default FilterBy;
