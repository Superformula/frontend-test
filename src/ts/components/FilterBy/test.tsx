// Imports
// -------

// Libraries
import React from 'react';
import renderer from 'react-test-renderer';
// Components
import FilterBy from './index';

// Mocks
// -----

jest.mock('../Button', () => 'Button');

// Internal
// --------

const children = "Text"

const getComponent = (props = {}) => renderer.create(
    <FilterBy {...props} >
        { children }
    </FilterBy>,
);

// Tests
// -----

describe('src/ts/components/FilterBy', () => {
    let component;

    describe('Rendering', () => {
        it('renders default correctly', () => {
            component = getComponent();
            expect(component).toMatchSnapshot();
        });
    });
});
