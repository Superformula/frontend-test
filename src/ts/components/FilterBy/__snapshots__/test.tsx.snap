// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/ts/components/FilterBy Rendering renders default correctly 1`] = `
<div
  className="filterBy"
>
  <div
    className="filterByText"
  >
    Filter By:
  </div>
  <div
    className="filterByOption"
  >
    Text
  </div>
  <div
    className="clearAllButton"
  >
    <Button
      disabled={false}
      onClick={[Function]}
      text="Clear All"
      tier="secondary"
    />
  </div>
</div>
`;
